File: scripts/fill_pdf.py
from pdfrw import PdfReader, PdfWriter, PdfDict
import json

data = json.load(open('data/this_week.json'))

fields = {
    'Date': data.get('date', ''),
    'Service Time': data.get('service_time', ''),
    'Lay Eucharistic Ministers': ', '.join(data.get('lay_eucharistic_ministers', [])),
    'Lay Readers': ', '.join(data.get('lay_readers', [])),
    'SCRIPTURE & PSALM': data.get('scripture_psalm', ''),
    'PRAYERS': data.get('prayers', ''),
    'Hospitality (Coffee/Donuts)': ', '.join(data.get('hospitality', [])),
    'Prayer Partners': ', '.join(data.get('prayer_partners', [])),
    'Ushers': ', '.join(data.get('ushers', [])),
    'Greeters': ', '.join(data.get('greeters', [])),
    'CRUCIFER': data.get('crucifer', ''),
    'TORCHBEARER_1': data.get('torchbearer_1', ''),
    'TORCHBEARER_2': data.get('torchbearer_2', ''),
    'TEAM LEAD': data.get('team_lead', ''),
}

pdf = PdfReader('pdfs/blank/WorshipTeam_Official.pdf')
for page in pdf.pages:
    if getattr(page, 'Annots', None):
        for annot in page.Annots or []:
            if annot.Subtype == '/Widget' and annot.T:
                name = annot.T[1:-1]
                if name in fields:
                    annot.V = PdfDict(v=str(fields[name]))
                    annot.AP = None

PdfWriter().write('pdfs/output/WorshipTeam_Filled.pdf', pdf)

File: scripts/fill_pdf.js
import fs from 'fs';
import { PDFDocument } from 'pdf-lib';

const data = JSON.parse(fs.readFileSync('data/this_week.json', 'utf8'));

const fieldText = {
  DATE: data.date,
  SERVICE_TIME: data.service_time,
  LEMS: (data.lay_eucharistic_ministers || []).join(', '),
  LAY_READERS: (data.lay_readers || []).join(', '),
  SCRIPTURE_PsALM: data.scripture_psalm || '',
  PRAYERS: data.prayers || '',
  HOSPITALITY: (data.hospitality || []).join(', '),
  PRAYER_PARTNERS: (data.prayer_partners || []).join(', '),
  USHERS: (data.ushers || []).join(', '),
  GREETERS: (data.greeters || []).join(', '),
  CRUCIFER: data.crucifer || '',
  TORCH1: data.torchbearer_1 || '',
  TORCH2: data.torchbearer_2 || '',
  TEAM_LEAD: data.team_lead || ''
};

async function main() {
  const pdfBytes = fs.readFileSync('pdfs/blank/WorshipTeam_Official.pdf');
  const pdfDoc = await PDFDocument.load(pdfBytes);
  const form = pdfDoc.getForm();

  // Map these names to the actual AcroForm field names in your PDF.
  const map = {
    DATE: 'Date',
    SERVICE_TIME: 'Service Time',
    LEMS: 'Lay Eucharistic Ministers',
    LAY_READERS: 'Lay Readers',
    SCRIPTURE_PsALM: 'SCRIPTURE & PSALM',
    PRAYERS: 'PRAYERS',
    HOSPITALITY: 'Hospitality (Coffee/Donuts)',
    PRAYER_PARTNERS: 'Prayer Partners',
    USHERS: 'Ushers',
    GREETERS: 'Greeters',
    CRUCIFER: 'CRUCIFER',
    TORCH1: 'TORCHBEARER_1',
    TORCH2: 'TORCHBEARER_2',
    TEAM_LEAD: 'TEAM LEAD'
  };

  for (const [key, fieldName] of Object.entries(map)) {
    try {
      const f = form.getTextField(fieldName);
      f.setText(fieldText[key] || '');
    } catch (e) {
      console.warn(`Field not found: ${fieldName}`);
    }
  }

  form.updateFieldAppearances();
  const filled = await pdfDoc.save();
  fs.writeFileSync('pdfs/output/WorshipTeam_Filled.pdf', filled);
}

main();